### PR TITLE
Update dependencies and package organization.

### PR DESCRIPTION
**Pull request by tgale96**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,15 +21,22 @@
 ]
 
 install_requires=[
-    'stanford-stk @ git+https://github.com/stanford-futuredata/stk.git@main',
-    'grouped_gemm @ git+https://github.com/tgale96/grouped_gemm@main',
-    'mosaicml-turbo==0.0.4',
+    "triton==2.1.0",
```
Comment (mvpatel2000) [inline setup.py:24]: does this need strict pin? Might cause problems down the road if someone else (eg torch) starts upgrading.... 

Could we lower bound?

File: setup.py [hunk 2]
```diff
@@ -21,15 +21,22 @@
...
+    "stanford-stk>=0.0.6",
 ]
 
 extra_deps = {}
 
-extra_deps['dev'] = [
-    'absl-py',
+extra_deps["gg"] = [
+    "grouped_gemm @ git+https://github.com/tgale96/grouped_gemm@main",
```
Comment (mvpatel2000) [inline setup.py:31]: We should use a tagged version instead of main branch

File: setup.py [hunk 3]
```diff
@@ -21,15 +21,22 @@
...
+]
+
+extra_deps["quant"] = [
+    "mosaicml-turbo==0.0.4",
```
Comment (mvpatel2000) [inline setup.py:35]:
```suggestion
    "mosaicml-turbo==0.0.5",
```
@dblalock  can we bump to 0.0.5?

**Review by mvpatel2000 — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (tgale96) [reply setup.py:24]: I'm concerned about Triton breaking us with a newer release. We could make it >=2.1.0, but that's a little risky.

File: setup.py (hunk 2 above)
Comment (tgale96) [reply setup.py:31]: I'm moving this package onto PyPi so we can push the new megablocks packages.

File: setup.py (hunk 3 above)
Comment (tgale96) [reply setup.py:35]: Lets do this in a separate PR.

Comment (dblalock) [reply setup.py:35]: +1, I'd rather not touch anything right now

**Pull request merged**